client/src/app/main-screen/main-screen.component.ts

```diff
@@ -0,0 +1,216 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { SessionService } from '../api/session/session.service';
+import { RequestService } from '../services/request.service';
+import { Select, TheeyeCredential } from '../common/global-constants';
+
+@Component({
+  selector: 'app-main-screen',
+  templateUrl: './main-screen.component.html',
+  styleUrls: ['./main-screen.component.less', './main-screen.component.css']
+})
+export class MainScreenComponent implements OnInit {
+
+	userEmail:string=''
+	userRole:string=''
+	userToken:string=''
+	userCustomer: string = ''
+
+	taskSelect:Select[]=[]
+	modeSelect:Select[]=[
+		{id:'sync',name:'API SYNC'},
+		{id:'supervisor', name:'SUPERVISOR'}
+	]
+	inputArguments:string=''
+	inputArgumentsArray:string[] = []
+	selectedTask:string=''
+	selectedMode:string='sync'
+	taskResultRaw:string=''
+	taskResultOutput:string=''
+	taskResult:{attr:string, value:string}[] = []
+	selectedTaskName: string = ''
+	taskResultInput: string = ''
+	showOverlay:boolean=true
+	showPretty:boolean=false
+	showRaw:boolean=false
+	jobId:string=''
+
+	constructor(private sessionService:SessionService, private router:Router, private requestService: RequestService) { }
+
+	getTasks = async ():Promise<any> => {
+		return new Promise((resolve, reject) => {
+			this.requestService.getTaskList(this.userToken, this.userCustomer)
+				.subscribe({
+					next: data => {
+						console.log('getTasks: Fetched res data')
+						resolve(data)
+					},
+					error: error => {
+						console.error('getTasks: Error requesting data!')
+						reject(error)
+					}
+				}
+			)
+		})
+	}
+
+	getTaskResult = async ():Promise<any> => {
+		return new Promise((resolve, reject) => {
+			this.requestService.getTaskResult(this.userToken, this.jobId, this.userCustomer)
+				.subscribe({
+					next: data => {
+						console.log('getTaskResult: Fetched res data')
+						resolve(data)
+					},
+					error: error => {
+						console.error('getTaskResult: Error requesting data!')
+						reject(error)
+					}
+				}
+			)
+		})
+	}
+
+	getCredentialData = async ():Promise<any> => {
+		return new Promise((resolve, reject) => {
+			this.requestService.getCredentialData(this.userToken)
+				.subscribe({
+					next: data => {
+						console.log('getCredentialData: Fetched res data')
+						resolve(data)
+					},
+					error: error => {
+						console.error('getCredentialData: Error requesting data!')
+						reject(error)
+					}
+				}
+			)
+		})
+	}
+
+	taskResultButton = async():Promise<any> => {
+		try {
+			const response = await this.getTaskResult()
+			this.taskResultRaw = response
+			this.setResultVars(response)
+			this.prettifyResult(response)
+			this.showOverlay = false
+		} catch(e:any) {
+			this.taskResultRaw = e
+			this.setResultVars(e.error)
+			this.prettifyResult(e)
+			this.showOverlay = false
+		}
+	}
```

Comment:
in general reactive actions are not required to be "async/await"-ables
actions in most cases are performed async and events will tell you **when** are accomplished 

I believe this can be written as follow 

```suggestion
	taskResultButton = () => {
		this.getTaskResult().then(response => {
			this.taskResultRaw = response
			this.setResultVars(response)
			this.prettifyResult(response)
			this.showOverlay = false
		}).catch( e => {
			this.taskResultRaw = e
			this.setResultVars(e.error)
			this.prettifyResult(e)
			this.showOverlay = false
		})
	}
```

if you are not returning any value , and the outer function is not waiting the result (I'm guessing the scenario) , then you don't need async functions here